fix: allow clients to set client synchronization mode distributed authority

NoelStephensUnity opened the pull request:
This update allows clients to set the client synchronization mode since any client can be promoted to session owner.

## Changelog

- Changed: Clients can set the `NetworkSceneManager` client synchronization mode when using a distributed authority network topology.

## Testing and Documentation

- No tests have been added.
- Documentation additions are necessary and [being tracked internally](https://docs.google.com/document/d/1yaljfx-ypRADhJ08mNpzXwFWsSJ1usqbQONfKrErZ34/edit)

<!--  Uncomment and mark items off with a * if this PR deprecates any API:
### Deprecated API
- [ ] An `[Obsolete]` attribute was added along with a `(RemovedAfter yyyy-mm-dd)` entry.
- [ ] An [api updater] was added.
- [ ] Deprecation of the API is explained in the CHANGELOG.
- [ ] The users can understand why this API was removed and what they should use instead.
-->

NoelStephensUnity committed:
fix

Allow clients to set client synchronization mode when in distributed authority mode since clients can be promoted to session owner.

NoelStephensUnity committed:
test

apply same rules for the integration test version of the scene handler

NoelStephensUnity committed:
update

adding change log entry

fluong6 commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/DefaultSceneManagerHandler.cs:
```diff
@@ -333,8 +333,9 @@ public void MoveObjectsFromSceneToDontDestroyOnLoad(ref NetworkManager networkMa
-            // Don't let non-authority set this value
-            if ((!networkManager.DistributedAuthorityMode && !networkManager.IsServer) || (networkManager.DistributedAuthorityMode && !networkManager.LocalClient.IsSessionOwner))
+            // In client-server, we don't let client's set this value.
+            // In dsitributed authority, since session owner can be promoted clients can set this value
```
minor typo: dsitributed

NoelStephensUnity replied:
:godmode: 
Nice cactch! Thank you!
👍

NoelStephensUnity committed:
style

fixing spelling typo.

NoelStephensUnity committed:
test

Adjusting for the modification.

NoelStephensUnity committed:
fix

Resolving issue where element index number might not be a 1:1 match based on an incrementing indices.

NoelStephensUnity committed:
fix

Reverse through the list to avoid collection modified exception.

NoelStephensUnity committed:
style

Remove an "auto-added" using directive...

NoelStephensUnity committed:
fix

Dumb mistake on the reverse iteration...greater than or equal to zero is the condition.
